persistent-postgresql/Database/Persist/Postgresql/JSON.hs

```diff
@@ -0,0 +1,146 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Database.Persist.Postgresql.JSON
+  ( (@>.)
+  , (<@.)
+  , Value()
+  ) where
+
+import Data.Aeson (FromJSON, ToJSON, Value, encode, eitherDecodeStrict)
+import qualified Data.ByteString.Lazy as BSL
+import Data.Proxy (Proxy)
+import Data.Text as T (Text, pack, concat)
+import Data.Text.Encoding as TE (encodeUtf8)
+
+import Database.Persist (EntityField, Filter(..), PersistValue(..), PersistField(..), PersistFilter(..))
+import Database.Persist.Sql (PersistFieldSql(..), SqlType(..))
+
+
+infix 4 @>., <@.
+
+-- | This operator checks inclusion of the JSON value
+-- on the right hand side in the JSON value on the left
+-- hand side.
+--
+-- Any empty Object matches any object
+-- @
+-- {"a":1,"b":false} @> {} --> True
+-- @
+--
+-- Any key-value will be matched top-level
+-- @
+-- {"a":1,"b":{"c:true"}} @> {"a":1}         --> True
+-- {"a":1,"b":{"c:true"}} @> {"b":1}         --> False
+-- {"a":1,"b":{"c:true"}} @> {"b":{}}        --> True
+-- {"a":1,"b":{"c:true"}} @> {"c":true}      --> False
+-- {"a":1,"b":{"c:true"}} @> {"b":{c":true}} --> True
+-- @
+--
+-- Any empty Array matches any array
+-- @
+-- [1,2,"hi",false,null] @> [] --> True
+-- @
+--
+-- Any array has to be a sub-set
+-- @
+-- [1,2,"hi",false,null] @> [1]                   --> True
+-- [1,2,"hi",false,null] @> [null,"hi"]           --> True
+-- [1,2,"hi",false,null] @> ["hi",true]           --> False
+-- [1,2,"hi",false,null] @> ["hi",2,null,false,1] --> True
+-- [1,2,"hi",false,null] @> [3,2,"hi"]            --> False
+-- @
+--
+-- For any other JSON values the `(@>.)` operator
+-- functions like an equivalence operator.
+-- @
+-- "hello" @> "hello" --> True
+-- "hello" @> "Hello" --> False
+-- "hello" @> "h"     --> False
+-- "hello" @> {"a":1} --> False
+-- "hello" @> []      --> False
+--
+-- 5       @> 5       --> True
+-- 5       @> 5.00    --> True
+-- 5       @> 1       --> False
+-- 5       @> 7       --> False
+-- 12345   @> 1234    --> False
+-- 12345   @> 2345    --> False
+-- 12345   @> "12345" --> False
+-- 12345   @> [1,2,3,4,5] --> False
+--
+-- true    @> true    --> True
+-- true    @> false   --> False
+-- false   @> true    --> False
+-- true    @> "true"  --> False
+--
+-- null    @> null    --> True
+-- null    @> 23      --> False
+-- null    @> "null"  --> False
+-- null    @> {}      --> False
+-- @
```

Comment:
I don' think these haddocks are rendering how you want them to

![image](https://user-images.githubusercontent.com/1274145/37161302-24247c6e-22a7-11e8-9ad0-0fb222f8e04e.png)